src/cmap/manager.rs

```diff
@@ -24,6 +24,22 @@ impl PoolManager {
         let _ = self.sender.send(PoolManagementRequest::Clear);
     }
 
+    /// Mark the pool as "ready" as per the CMAP specification.
+    pub(super) async fn mark_as_ready(&self) {
+        let (sender, receiver) = tokio::sync::oneshot::channel();
+        if self
+            .sender
+            .send(PoolManagementRequest::MarkAsReady {
+                completion_handler: sender,
```

Comment:
We use a completion handler here because we don't want to mark the server as known before we know the pool is ready, otherwise we could select a server whose pool can't process  `check_out` requests.